ggshield/cmd/hmsl/check_secret_manager/hashicorp_vault.py

```diff
@@ -0,0 +1,97 @@
+import os
+from typing import Any
+
+import click
+
+from ggshield.cmd.common_options import add_common_options
+from ggshield.verticals.hmsl.secret_manager.hashicorp_vault import (
+    VaultCliTokenFetchingError,
+    get_vault_cli_token,
+)
+
+
+def _get_vault_token(use_vault_cli_token: bool) -> str:
+    """
+    Get the Vault token to use.
+    """
+    if use_vault_cli_token:
+        try:
+            return get_vault_cli_token()
+        except VaultCliTokenFetchingError as exc:
+            raise click.UsageError(
+                f"could not get the token from Vault CLI: {str(exc)}"
+            ) from exc
+
+    if "VAULT_TOKEN" in os.environ:
+        return os.environ["VAULT_TOKEN"]
+
+    raise click.UsageError(
+        "you need to specify the Vault token to use, either through the VAULT_TOKEN"
+        " environment variable or by using --use-cli-token to use the token "
+        "from the Vault CLI."
+    )
+
+
+@click.command(hidden=True)
+@click.option(
+    "--use-cli-token",
+    "use_cli_token",
+    is_flag=True,
+    show_default=True,
+    default=False,
+    help="Instead of getting the token from the an environment variable, "
+    "get the token from the CLI tool.",
```

Comment:
nit: typo and repetition.
```suggestion
    help="Instead of getting the token from the environment variable, "
    "get it from the CLI tool.",
```